File: ja-muslims-directory/frontend/types/masjid.ts
export enum MasjidType {
  MASJID = 'masjid',
  MUSALLA = 'musalla',
}

export interface MemberBasic {
  id: number;
  muslim_name: string;
  legal_name: string;
}

export interface Masjid {
  id: number;
  name: string;
  type: MasjidType;
  address: string;
  city?: string;
  parish: string;
  postal_code?: string;
  phone?: string;
  email?: string;
  website?: string;
  imam_id?: number;
  established_year?: number;
  capacity?: number;
  facilities?: string;
  prayer_times_info?: string;
  jummah_time?: string;
  activities?: string;
  created_at: string;
  updated_at?: string;
}

export interface MasjidWithRelations extends Masjid {
  imam?: MemberBasic;
  shura_members: MemberBasic[];
  affiliated_members_count?: number;
}

export interface MasjidFormData {
  name: string;
  type: MasjidType;
  address: string;
  city?: string;
  parish: string;
  postal_code?: string;
  phone?: string;
  email?: string;
  website?: string;
  imam_id?: number;
  established_year?: number;
  capacity?: number;
  facilities?: string;
  prayer_times_info?: string;
  jummah_time?: string;
  activities?: string;
  shura_member_ids?: number[];
}

// Jamaican parishes (same as in restaurant types)
export const PARISHES = [
  'Kingston',
  'St. Andrew',
  'St. Thomas',
  'Portland',
  'St. Mary',
  'St. Ann',
  'Trelawny',
  'St. James',
  'Hanover',
  'Westmoreland',
  'St. Elizabeth',
  'Manchester',
  'Clarendon',
  'St. Catherine',
] as const;

File: ja-muslims-directory/backend/app/schemas/masjid.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime
from app.models.masjid import MasjidType


class MasjidBase(BaseModel):
    name: str
    type: MasjidType
    address: str
    city: Optional[str] = None
    parish: str
    postal_code: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    website: Optional[str] = None
    imam_id: Optional[int] = None
    established_year: Optional[int] = None
    capacity: Optional[int] = None
    facilities: Optional[str] = None
    prayer_times_info: Optional[str] = None
    jummah_time: Optional[str] = None
    activities: Optional[str] = None


class MasjidCreate(MasjidBase):
    shura_member_ids: Optional[List[int]] = []


class MasjidUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[MasjidType] = None
    address: Optional[str] = None
    city: Optional[str] = None
    parish: Optional[str] = None
    postal_code: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    website: Optional[str] = None
    imam_id: Optional[int] = None
    established_year: Optional[int] = None
    capacity: Optional[int] = None
    facilities: Optional[str] = None
    prayer_times_info: Optional[str] = None
    jummah_time: Optional[str] = None
    activities: Optional[str] = None
    shura_member_ids: Optional[List[int]] = None


class MemberBasic(BaseModel):
    id: int
    muslim_name: str
    legal_name: str
    
    class Config:
        from_attributes = True


class Masjid(MasjidBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True


class MasjidWithRelations(Masjid):
    imam: Optional[MemberBasic] = None
    shura_members: List[MemberBasic] = []
    affiliated_members_count: Optional[int] = 0
    
    class Config:
        from_attributes = True